### PR TITLE
add Azure DevOps to required permissions list

### PR DESCRIPTION
**Pull request by khorne3**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A technical writer reviews the content or PR

### CODE REVIEW

File: docs/deployment/deployment-checklist.md [hunk 1]
```diff
@@ -111,9 +111,10 @@ The following checklist breaks down permissions required by Semgrep features.
 <Tabs
     defaultValue="gh"
     values={[
-    {label: 'GitHub', value: 'gh'},
```
Comment (s-santillan) [inline docs/deployment/deployment-checklist.md:114]: Silly question from me - should we group by popularity or by alphabetization? I'm sure at least 50% of the time people will need to click the GH tab 😓 They are the most popular.

File: docs/deployment/deployment-checklist.md [hunk 2]
```diff
@@ -167,6 +168,16 @@ The following checklist breaks down permissions required by Semgrep features.
 
 </TabItem>
 
+<TabItem value='az'>
+
+#### Azure DevOps
+
+| Feature  | Permission |
+| -------  | -------  |
+| Pull request (PR) comments.  | Able to create [user personal access tokens](https://learn.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate). |
```
Comment (s-santillan) [inline docs/deployment/deployment-checklist.md:177]: Do we have any enterprise level access tokens? Or are the user personal access tokens applicable for orgs?

File: docs/deployment/deployment-checklist.md (hunk 1 above)
Comment (khorne3) [reply docs/deployment/deployment-checklist.md:114]: I alphabetized this due to my neurosis, but I left GH as the default, so it should be showing most of the time anyway; if you think it should be by popularity, I don't mind changing it back!

File: docs/deployment/deployment-checklist.md (hunk 2 above)
Comment (khorne3) [reply docs/deployment/deployment-checklist.md:177]: Just PAT (Azure seems to have only two kinds: PAT and pipeline-related)

**Review by s-santillan — APPROVED** [review]

File: docs/deployment/deployment-checklist.md (hunk 1 above)
Comment (s-santillan) [reply docs/deployment/deployment-checklist.md:114]: @khorne3 I didn't notice, thank you - yeah GH is the default, I should have tested first.

**Pull request merged**